cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceProgressUtils.java

```diff
@@ -0,0 +1,80 @@
+     *     - The data movement rate is zero, making the time to completion estimation impossible.
+     */
+    /* test */ static int estimateTimeToCompletionInMinutes(long taskStartTime,
+                                                        long currentTime,
+                                                        int totalDataToMove,
+                                                        int finishedDataMovement)
+            throws IllegalArgumentException, ArithmeticException {
+        if (taskStartTime < 0 || currentTime < 0 || totalDataToMove < 0 || finishedDataMovement < 0) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid negative value(s) provided for one of the following arguments: taskStartTime: %d, currentTime: %d, totalDataToMove: %d, finishedDataMovement: %d.",
+                    taskStartTime, currentTime, totalDataToMove, finishedDataMovement)
+            );
+        }
+        if (!(currentTime >= taskStartTime)) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid value(s) provided for the arguments taskStartTime: %d, currentTime: %d. " +
+                            "The value of taskStartTime must be greater than the value  currentTime", taskStartTime, currentTime)
+            );
+        }
+        // Calculate the seconds between the start time and the current time
+        long timeElapsedInSeconds = currentTime - taskStartTime;
+        if (timeElapsedInSeconds == 0) {
+            throw new ArithmeticException("Time elapsed less than or equal to zero, cannot calculate byte movement rate.");
+        }
+
+        double rate = (double) finishedDataMovement / timeElapsedInSeconds;
+        if (rate == 0) {
+            throw new ArithmeticException("Byte movement rate is zero, cannot estimate time to completion.");
+        }
+
+        int dataLeftToMove = totalDataToMove - finishedDataMovement;
+        return (int) (dataLeftToMove / (rate * SECONDS_IN_A_MINUTE));
+    }
+
+    /**
+     * Estimates the percentage of data movement completed of an ongoing partition rebalance given the total number of
+     * megabytes to be moved and the number of megabytes already moved as part of the rebalance.
+     *
+     * @param totalDataToMove The total amount of data that needs to be moved, in megabytes.
+     * @param finishedDataMovement The amount of data that has already been moved, in megabytes.
+     * @return The percentage of data movement completed as a rounded down integer between [0 and 100].
+     * @throws IllegalArgumentException if `finishedDataMovement` is greater than `totalDataToMove` or if either value is negative.
+     */
+    /* test */ static int estimateCompletedByteMovementPercentage(int totalDataToMove, int finishedDataMovement) {
+        if (!(finishedDataMovement <= totalDataToMove) || finishedDataMovement < 0) {
```

Comment:
In general my feeling, as before, is that negative conditions made the code less readable unless you hide the condition within a dedicated and well named method. But I would go with just avoiding a negative condition here.

Reply:
I agree, why not:
```suggestion
        if ( finishedDataMovement > totalDataToMove || finishedDataMovement < 0) {
```

Reply:
Fixed.